### ISSUE TITLE
The getter 'store' was called on null

### ISSUE THREAD

**Issue by darioielardi** [brianegan/flutter_redux#38]
Hi,

I'm trying to implement redux on my flutter app, versions:
```
  redux: "^3.0.0"
  flutter_redux: "^0.5.0"
```

When I run my app the error in the title is thrown.
I followed the Dart 2 migration guide for both the packages.

Flutter: 0.2.8
Dart: 2.0.0

**Comment by brianegan** [comment]
Sorry about that! I need to add one more error case to the error message: How are you importing your files? If you're using relative paths, e.g. `import '../store.dart'` rather than `import package:my_app/store.dart` you may run into this issue.

If that's not the case, could you please provide a bit more code, even a repo with the code that's failing? That way I can help debug :)

**Comment by darioielardi** [comment]
Thanks @brianegan for your answer, but trying to import with absolute path didn't change anything, the error message is still there. 
Unfortunately I can't provide the repo, but basically my code looks like:
```
import 'package:test/modules/store.dart';

  runApp(
    new StoreProvider<AppState>(
      store: store,
      child: new MaterialApp(
        title: 'Test App',
        theme: new ThemeData(
          textTheme: new TextTheme(
            display1: new TextStyle(fontFamily: 'Rubik'),
          )
        ),
        home: new TestApp()
      )
    )
  );
```
my store:
```
var store = new Store<AppState>(
  rootReducer,
  initialState: new AppState(
    user: new UserState(null, true),
    connection: new ConnectionState(false)
  ),
  middleware: middlewares
);
```
and in my TestApp basically there's a StoreConnector that causes the error, nothing happens if I avoid to use that.

The error:

```
I/flutter (11060): The following NoSuchMethodError was thrown building StoreConnector<AppState, UserState>(dirty):
I/flutter (11060): The getter 'store' was called on null.
I/flutter (11060): Receiver: null
I/flutter (11060): Tried calling: store
```

**Comment by brianegan** [comment]
Hrm, interesting -- could you please check your `pubspec.lock` and verify you're on `flutter_redux` version 0.5.0? Since 0.4.1, if the store is null, it should throw a different error message from this.

Thanks!

**Comment by darioielardi** [comment]
Thanks again for your help, looks like I'm on the correct version:
```
  flutter_redux:
    dependency: "direct main"
    description:
      name: flutter_redux
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.5.0"
```

**Comment by brianegan** [comment]
Hrm, this is odd... I can't quite reproduce the issue, and it's a bit baffling that you're getting that particular error message, since I explicitly do a null-check before accessing the store :(

Does the example app in this repo run for you?

**Comment by darioielardi** [comment]
The example in this repo works correctly.
Trying to call `StoreProvider.of(context)` in the build method an error is thrown but with `StoreProvider.of<AppState>(context)` everything works.

**Comment by darioielardi** [comment]
Now the error is different. My code:
```
    return new StoreConnector<AppState, UserState>(
      converter: (store) => store.state.user,
      builder: (BuildContext ctx, userState) {
        ...
      }
    );
```
It throws this error:
```
The following StoreProviderError was thrown building StoreConnector<AppState, UserState>(dirty):
Error: No StoreProvider<AppState> found. To fix, please try:

  * Using Dart 2 (required) by using the --preview-dart-2 flag
  * Wrapping your MaterialApp with the StoreProvider<State>, rather than an individual Route
  * Providing full type information to your Store<State>, StoreProvider<State> and
     StoreConnector<State, ViewModel>
```
Which is quite weird because my StoreProvider is implemented like:
```
final store = new Store<AppState>(...);
...
return new StoreProvider<AppState>(
  store: store,
  child: new MaterialApp(...)
);
```

**Comment by darioielardi** [comment]
In some way I solved the problem.

Basically I created another project, and copy-pasting the code the same error was thrown, but when I tried to rewrite it (without referring at the previous version) everything worked. Then I figured out that it depended on the imports, as @brianegan told me some comments above, so using only relatives imports everything was okay, so I tried to replace every absolute import on the previous project with a relative one, but it didn't work, probably due to some other messy stuff. Now I'm working on the second project.
